### PR TITLE
Feature/issues/642

### PR DESCRIPTION
**Pull request by Mizzick**
#642

### CODE REVIEW

File: Extension/browser/safari/lib/converter.js [hunk 1]
```diff
@@ -769,10 +769,8 @@ var SafariContentBlockerConverter = {
                 } else if (item.action.type == 'ignore-previous-rules' && 
                     (ruleText && ruleText.indexOf('generichide') > 0)) {
                     contentBlocker.cssBlockingGenericHideExceptions.push(item);
-                } else if (item.action.type == 'ignore-previous-rules' && 
-                        (item.trigger["resource-type"] && 
-                        item.trigger["resource-type"].length > 0 &&
-                        item.trigger["resource-type"][0] == 'document')) {
+                } else if (item.action.type == 'ignore-previous-rules' &&
+                    (ruleText && ruleText.indexOf('elemhide') > 0)) {
```
Comment (ameshkov) [inline Extension/browser/safari/lib/converter.js]: Make it stronger, check that rule endsWith `$elemhide`

Comment (Mizzick) [inline Extension/browser/safari/lib/converter.js]: ok, fixed this an other way.

File: Extension/browser/safari/lib/converter.js [hunk 2]
```diff
@@ -442,7 +442,29 @@ var SafariContentBlockerConverter = {
      */
     convertLine: function (ruleText, errors) {
```
Comment (atropnikov) [inline Extension/browser/safari/lib/converter.js]: IDE shows me that this method is unused.

File: Extension/browser/safari/lib/converter.js [hunk 3]
```diff
@@ -453,10 +475,9 @@ var SafariContentBlockerConverter = {
                 throw new Error('Cannot create rule from: ' + ruleText);
             }
 
-            return this._convertAGRule(agRule);
-
+            return agRule;
         } catch (ex) {
-            var message = 'Error converting rule from: ' + ruleText + ' cause:\n' + ex;
+            var message = 'Error creating rule from: ' + ruleText + ' cause:\n' + ex;
```
Comment (atropnikov) [inline Extension/browser/safari/lib/converter.js]: We can use formatted strings in `adguard.console.debug`.

File: Extension/browser/safari/lib/converter.js [hunk 4]
```diff
@@ -442,7 +442,29 @@ var SafariContentBlockerConverter = {
...
+     * Creates AG rule form text
+     *
+     * @param ruleText
+     * @param errors
+     */
+    parseAGRule: function (ruleText, errors) {
+        try {
+            if (ruleText === null || ruleText === '' ||
                 ruleText.indexOf('!') === 0 || ruleText.indexOf(' ') === 0 ||
                 ruleText.indexOf(' - ') > 0) {
```
Comment (atropnikov) [inline Extension/browser/safari/lib/converter.js]: I don't quite understand why do you miss the rule that starts with space or contains `-`?

File: Extension/tests/safari-converter/test-safari-converter.js [hunk 5]
```diff
@@ -383,4 +383,33 @@ QUnit.test("UpperCase domains", function (assert) {
...
+
+    var converted = JSON.parse(result.converted);
+    assert.equal(4, converted.length);
+
+    assert.equal(converted[0].action["selector"], "#root > section.b-header.b-header-main.js-header:nth-child(4) > div.g-layout > div.row");
+    assert.equal(converted[0].action.type, "css-display-none");
+
+    assert.equal(converted[1].trigger["url-filter"], ".*");
+    assert.equal(converted[1].trigger["if-domain"], "*lenta.ru");
+    assert.equal(converted[1].trigger["url-filter"], ".*");
```
Comment (atropnikov) [inline Extension/tests/safari-converter/test-safari-converter.js]: Duplicate

File: Extension/browser/safari/lib/converter.js (hunk 2 above)
Comment (Mizzick) [inline Extension/browser/safari/lib/converter.js]: As I know it's used in iOS

Comment (atropnikov) [inline Extension/browser/safari/lib/converter.js]: Ok. Please, add comment about it.

File: Extension/browser/safari/lib/converter.js (hunk 4 above)
Comment (Mizzick) [inline Extension/browser/safari/lib/converter.js]: It's all about comments and some strange lines in filters files.

File: Extension/browser/safari/lib/converter.js (hunk 3 above)
Comment (Mizzick) [inline Extension/browser/safari/lib/converter.js]: yeap, but it's also collected in `errors` collection.

File: Extension/tests/safari-converter/test-safari-converter.js (hunk 5 above)
Comment (Mizzick) [inline Extension/tests/safari-converter/test-safari-converter.js]: ok

**Pull request merged**